app/controllers/api/v1/messages_controller.rb

```diff
@@ -51,7 +47,6 @@ def show
       api :POST, "/v1/messages", "Create an message"
       param_group :message
       def create
-        @message.order_id = order_id
         @message.sender_id = current_user.id
```

Comment:
To maintain backwards compatibility with the `item_id, order_id, offer_id` on old apps, create the following function and then call it in the `create` and `update` methods (if it exists)

```
def create
...
  if !@message.messageable_id
    @message.messageable_type, @message.messagable_id = old_messagesable_params
  end
...
end
```
and then in the controller, a method to manage the params

```
# for backwards compatibility
def old_messagesable_params
  case
  when params["item_id"].present?
    ['Item', params["item_id"]]
  when params["offer_id"].present?
    ['Offer', params["offer_id"]]
  when params["order_id"].present?
    ['Order', params["order_id"]]
  else
    [nil,nil]
  end
end
```

Don't forget to add some specs for this method too

Reply:
I have updated. Had to make manipulation to the params, as the abilities will not allow to create message. [This commit](https://github.com/crossroads/api.goodcity/pull/999/commits/0fa396579e35c6ea634fbbe50faeef1a94480f6a#diff-e34206da28447c019a2c53781f9bbdbaR110) has the changes